src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenNullCheckedParameterTests.cs

```diff
@@ -1262,6 +1177,348 @@ .maxstack  1
+    IL_0008:  newobj     ""System.ArgumentNullException..ctor(string)""
+    IL_000d:  throw
+    IL_000e:  ldc.i4.s   -2
+    IL_0010:  newobj     ""Iterators.<<Use>g__GetChars|0_0>d..ctor(int)""
+    IL_0015:  dup
+    IL_0016:  ldarg.0
+    IL_0017:  stfld      ""string Iterators.<<Use>g__GetChars|0_0>d.<>3__s""
+    IL_001c:  ret
+}");
+        }
+
+        [Fact]
+        public void TestNullCheckedEnumeratorInLocalFunction()
+        {
+            var source = @"
+using System.Collections.Generic;
+class Iterators
+{
+    void Use()
+    {
+        IEnumerator<char> e = GetChars(""hello"");
+        IEnumerator<char> GetChars(string s!)
+        {
+            foreach (var c in s)
+            {
+                yield return c;
+            }
+        }
+    }
+
+}";
+            var compilation = CompileAndVerify(source);
+            compilation.VerifyIL("Iterators.<Use>g__GetChars|0_0(string)", @"
+{
+    // Code size       28 (0x1c)
+    .maxstack  3
+    IL_0000:  ldarg.0
+    IL_0001:  brtrue.s   IL_000e
+    IL_0003:  ldstr      ""s""
+    IL_0008:  newobj     ""System.ArgumentNullException..ctor(string)""
+    IL_000d:  throw
+    IL_000e:  ldc.i4.0
+    IL_000f:  newobj     ""Iterators.<<Use>g__GetChars|0_0>d..ctor(int)""
+    IL_0014:  dup
+    IL_0015:  ldarg.0
+    IL_0016:  stfld      ""string Iterators.<<Use>g__GetChars|0_0>d.s""
+    IL_001b:  ret
+}");
+        }
+
+        [Fact]
+        public void TestNullCheckedIteratorExecution()
+        {
+            var source = @"
+using System;
+using System.Collections.Generic;
+class Program
+{
+    IEnumerable<char> GetChars(string s!)
+        {
+            foreach (var c in s)
+            {
+                yield return c;
+            }
+    }
+    static void Main()
+    {
+        Program p = new Program();
+        var c = Invoke(() => p.GetChars(string.Empty));
+        var e = Invoke(() => c.GetEnumerator());
+        Invoke(() => e.MoveNext());
+        c = Invoke(() => p.GetChars(null));
+    }
+    static T Invoke<T>(Func<T> f)
+    {
+        T t = default;
+        try
+        {
+            t = f();
+            Console.WriteLine(""ok"");
+        }
+        catch (Exception e)
+        {
+            Console.WriteLine(e.GetType());
+        }
+        return t;
+    }
+}";
+            CompileAndVerify(source, expectedOutput: @"
+ok
+ok
+ok
+System.ArgumentNullException");
+        }
+
+        [Fact]
+        public void TestNullCheckedWithMissingType()
+        {
+            var source =
+@"
+class Program
+{
+    static void Main(string[] args!) { }
+}";
+            var comp = CreateCompilation(source);
+            comp.MakeMemberMissing(WellKnownMember.System_ArgumentNullException__ctorString);
+            comp.MakeTypeMissing(WellKnownType.System_ArgumentNullException);
+            comp.VerifyEmitDiagnostics(
+                    // (4,31): error CS0656: Missing compiler required member 'System.ArgumentNullException..ctor'
```

Comment:
>Missing compiler required member 'System.ArgumentNullException..ctor' [](start = 45, length = 69)

Is this error reported by binding or lowering? If binding (`ParameterHelpers.AddNullCheckingErrorsToParameter`), consider changing each of these tests to `VerifyDiagnostics`.